lib/active_record/typed_store/extension.rb

```diff
@@ -12,15 +12,27 @@ def typed_store(store_attribute, options={}, &block)
       unless self < Behavior
         include Behavior
         class_attribute :typed_stores, :store_accessors, instance_accessor: false
+
+        def inherited(sub_class)
+          super(sub_class)
+
+          sub_class.typed_stores = Marshal.load(Marshal.dump(self.typed_stores)) if self.respond_to? :typed_stores
```

Comment:
`Marshal.load(Marshal.dump())` isn't acceptable here, and would break if the code contain procs and things like that.

We should dup the necessary things, not rely on a deep_dup hack.

Reply:
I'll see if I can find another way to dup the stores, this is to prevent modifying the store in parent class